python_modules/dagster/dagster/_core/execution/context/data_version_cache.py

```diff
@@ -29,11 +29,14 @@
     )
     from dagster._core.event_api import EventLogRecord
     from dagster._core.events import DagsterEventType
+    from dagster._core.storage.event_log.base import AssetRecord
 
 
 if TYPE_CHECKING:
     from dagster._core.execution.context.compute import StepExecutionContext
 
+ASSET_RECORD_BATCH_SIZE = 100
```

Comment:
Is there a way we can make this configurable and driven from a source of truth in Dagster Plus when the user is a Plus user? It would be great to have control over this to tune perf and control incidents.

Reply:
I think the main way to do this (since this called from user code) is to make it a property of the instance.  

something like:
```
# in DagterInstance
@property
def max_recommended_batch_size() -> int:
    ...
```

and then the question is how narrowly we scope it.  Is it for fetching events?  all records?

I guess the thing to do for this particular case is to have the recommended batch size set and then warn in OSS if it exceeds it.

Reply:
Yeah that makes sense. @gibsondan do you have a position on doing this approach?

Reply:
i think the limit here should probably be tuned to this specific query callsite (as opposed to perscribing the max amount of asset records you would want to fetch in any situation - it probably depends on the context)

I think it depends on whether we want to be able to tune it remotely on the cloud servers without the user taking any action. 

If we do, a simple option would be to make it an env var - this is how we handled it for batch writing store events, and there are various options for both users and for us to do remote tuning of env var values

Reply:
(if we do this we should just be sure to re-fetch the env var from os.environ every time it is accessed)

Reply:
Is this true even though this code is run in user-code (e.g. run containers)?

Reply:
I DMed daniel this exact question :-)